### PR TITLE
[zh-CN]: add translation for Using the Document Picture-in-Picture API

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

[Using the Document Picture-in-Picture API](https://developer.mozilla.org/en-US/docs/Web/API/Document_Picture-in-Picture_API/Using)

### Related issues and pull requests

part of the https://github.com/mdn/translated-content/issues/23557

### CODE REVIEW

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+l10n:
+  sourceCommit: d0b23f3f26637aa405ee9ee0a0892fc6e9b742ef
+---
+
+{{SeeCompatTable}}{{DefaultAPISidebar("Document Picture-in-Picture API")}}{{securecontext_header}}
+
+本指南提供了 {{domxref("Document Picture-in-Picture API", "Document Picture-in-Picture API", "", "nocode")}} 典型用法的详细指导。
+
+> [!NOTE]
+> 你可以在 [Document Picture-in-Picture API 示例](https://mdn.github.io/dom-examples/document-picture-in-picture/)中看到特色演示（另请参阅完整的 [源代码](https://github.com/mdn/dom-examples/tree/main/document-picture-in-picture)）。
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:13]:
```suggestion
> 你可以在 [Document Picture-in-Picture API 示例](https://mdn.github.io/dom-examples/document-picture-in-picture/)中看到特色演示（另请参阅完整的[源代码](https://github.com/mdn/dom-examples/tree/main/document-picture-in-picture)）。
```

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      pipMuteButton.textContent = "已静音";
+    }
+  });
+
+  pipWindow.document.body.append(pipMuteButton);
+});
+```
+
+> [!NOTE]
+> {{domxref("DocumentPictureInPictureEvent")}} 事件对象包含一个 `window` 属性，用于访问画中画窗口。
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:180]:
```suggestion
> **备注：** {{domxref("DocumentPictureInPictureEvent")}} 事件对象包含一个 `window` 属性，用于访问画中画窗口。
```

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+  document.getElementById("controlbar").appendChild(togglePipButton);
+}
+```
+
+如果可用，我们将删除“Document Picture-in-Picture API 不可用”消息，并添加 {{htmlelement("button")}} 元素以在文档画中画窗口中打开视频播放器。
+
+## 打开画中画窗口
+
+以下 JavaScript 调用 {{domxref("DocumentPictureInPicture.requestWindow", "window.documentPictureInPicture.requestWindow()")}} 打开一个空白的画中画窗口。返回的 {{jsxref("Promise")}} 用画中画 {{domxref("Window")}} 对象实现。使用 {{domxref("Element.append()")}} 将视频播放器移动到该窗口，然后我们显示消息通知用户它已被移动。
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:70]:
```suggestion
以下 JavaScript 调用 {{domxref("DocumentPictureInPicture.requestWindow", "window.documentPictureInPicture.requestWindow()")}} 打开一个空白的画中画窗口。返回的 {{jsxref("Promise")}} 兑现画中画 {{domxref("Window")}} 对象。使用 {{domxref("Element.append()")}} 将视频播放器移动到该窗口，然后我们显示消息通知用户它已被移动。
```

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+l10n:
+  sourceCommit: d0b23f3f26637aa405ee9ee0a0892fc6e9b742ef
+---
+
+{{SeeCompatTable}}{{DefaultAPISidebar("Document Picture-in-Picture API")}}{{securecontext_header}}
+
+本指南提供了 {{domxref("Document Picture-in-Picture API", "Document Picture-in-Picture API", "", "nocode")}} 典型用法的详细指导。
+
+> [!NOTE]
+> 你可以在 [Document Picture-in-Picture API 示例](https://mdn.github.io/dom-examples/document-picture-in-picture/)中看到特色演示（另请参阅完整的[源代码](https://github.com/mdn/dom-examples/tree/main/document-picture-in-picture)）。
```
Comment (jasonren0403) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:13]:
```suggestion
> 你可以在 [Document Picture-in-Picture API 示例](https://mdn.github.io/dom-examples/document-picture-in-picture/)中看到特性演示（另请参阅完整的[源代码](https://github.com/mdn/dom-examples/tree/main/document-picture-in-picture)）。
```

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        Document Picture-in-Picture API 不可用
+      </p>
+
+      <p></p>
+    </div>
+  </div>
+</div>
+```
+
+## 特征检测
```
Comment (jasonren0403) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:50]:
```suggestion
## 特性检测
```

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+如果可用，我们将删除“Document Picture-in-Picture API 不可用”消息，并添加 {{htmlelement("button")}} 元素以在文档画中画窗口中打开视频播放器。
+
+## 打开画中画窗口
+
+以下 JavaScript 调用 {{domxref("DocumentPictureInPicture.requestWindow", "window.documentPictureInPicture.requestWindow()")}} 打开一个空白的画中画窗口。返回的 {{jsxref("Promise")}} 兑现画中画 {{domxref("Window")}} 对象。使用 {{domxref("Element.append()")}} 将视频播放器移动到该窗口，然后我们显示消息通知用户它已被移动。
+
+`requestWindow()` 的 `width` 和 `height` 选项将画中画窗口设置为所需大小。如果选项值太大或太小而无法适应用户友好的窗口大小，浏览器可能会限制选项值。
+
+```js
```
Comment (jasonren0403) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:74]: 代码示例需要更新下

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+    link.type = styleSheet.type;
+    link.media = styleSheet.media;
+    link.href = styleSheet.href;
+    pipWindow.document.head.appendChild(link);
+  }
+});
+
+// ...
+```
+
```
Comment (jasonren0403) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:93]: 少了一章节“将样式表复制到画中画窗口中”

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+@media (display-mode: picture-in-picture) {
+  body {
+    background: red;
+  }
+}
+```
+
+仅当以画中画模式显示时，此代码片段才会将文档 `<body>` 的背景变为红色。
+
+在[我们的演示](https://mdn.github.io/dom-examples/document-picture-in-picture/)中，我们将 `display-mode: picture-in-picture` 值与 {{cssxref("@media/prefers-color-scheme", "prefers-color-scheme")}} 媒体功能相结合，以根据用户的配色方案偏好创建仅当应用以画中画模式显示时才应用的明暗配色方案。
```
Comment (jasonren0403) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:108]:
```suggestion
在[我们的演示](https://mdn.github.io/dom-examples/document-picture-in-picture/)中，我们将 `display-mode: picture-in-picture` 值与 {{cssxref("@media/prefers-color-scheme", "prefers-color-scheme")}} 媒体特性相结合，以根据用户的配色方案偏好创建仅当应用以画中画模式显示时才应用的明暗配色方案。
```

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+但是，你还需要考虑用户通过按下窗口本身上浏览器提供的关闭（X）按钮来关闭画中画窗口的情况。你可以通过使用 [`pagehide`](/zh-CN/docs/Web/API/Window/pagehide_event) 事件检测窗口何时关闭来处理这种情况：
+
+```js
+pipWindow.addEventListener("pagehide", (event) => {
+  inPipMessage.style.display = "none";
+  playerContainer.append(videoPlayer);
+});
+```
+
+## 当网站进入画中画模式时监听
```
Comment (jasonren0403) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:149]:
```suggestion
## 监听网站何时进入画中画模式
```

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+
+监听 `DocumentPictureInPicture` 实例上的 {{domxref("DocumentPictureInPicture.enter_event", "enter")}} 事件，了解画中画窗口何时打开。
+
+在我们的演示中，我们使用 `enter` 事件向画中画窗口添加静音切换按钮：
+
+```js
+documentPictureInPicture.addEventListener("enter", (event) => {
+  const pipWindow = event.window;
+  console.log("视频播放器已进入 pip 窗口");
```
Comment (jasonren0403) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:158]:
```suggestion
  console.log("视频播放器已进入画中画窗口");
```

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+  const pipMuteButton = pipWindow.document.createElement("button");
+  pipMuteButton.textContent = "已静音";
+  pipMuteButton.addEventListener("click", () => {
+    const pipVideo = pipWindow.document.querySelector("#video");
+    if (!pipVideo.muted) {
+      pipVideo.muted = true;
+      pipMuteButton.textContent = "未静音";
+    } else {
+      pipVideo.muted = false;
+      pipMuteButton.textContent = "已静音";
```
Comment (jasonren0403) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:169]:
```suggestion
      pipMuteButton.textContent = "静音";
```

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+
+在我们的演示中，我们使用 `enter` 事件向画中画窗口添加静音切换按钮：
+
+```js
+documentPictureInPicture.addEventListener("enter", (event) => {
+  const pipWindow = event.window;
+  console.log("视频播放器已进入 pip 窗口");
+
+  const pipMuteButton = pipWindow.document.createElement("button");
+  pipMuteButton.textContent = "已静音";
```
Comment (jasonren0403) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:161]:
```suggestion
  pipMuteButton.textContent = "静音";
```

File: files/zh-cn/web/api/document_picture-in-picture_api/using/index.md [hunk 13]
```diff
@@ -0,0 +1,196 @@
...
+  pipMuteButton.addEventListener("click", () => {
+    const pipVideo = pipWindow.document.querySelector("#video");
+    if (!pipVideo.muted) {
+      pipVideo.muted = true;
+      pipMuteButton.textContent = "未静音";
```
Comment (jasonren0403) [inline files/zh-cn/web/api/document_picture-in-picture_api/using/index.md:166]:
```suggestion
      pipMuteButton.textContent = "取消静音";
```

**Review by jasonren0403 — CHANGES_REQUESTED** [review]
存在一些问题

**Review by jasonren0403 — APPROVED** [review]
👍

**Pull request merged**